src/chatd.cpp

```diff
@@ -744,7 +749,8 @@ HistSource Chat::getHistoryFromDbOrServer(unsigned count)
 
 void Chat::requestHistoryFromServer(int32_t count)
 {
-    assert(mConnection.isOnline());
+    // the connection must be established (for a JOIN + HIST), but not yet logged in
+    assert(mConnection.state() >= Connection::kStateConnected);
```

Comment:
History from server is requested in 2 code paths, one when there's a connection already in place and it's logged in, but the other path is when the connection is alive but not yet logged in (as the comment states at the end of the line). So I consider the comment as correct.